gems/manageiq-providers-amazon/app/models/manageiq/providers/amazon/cloud_manager.rb

```diff
@@ -16,13 +14,41 @@ class ManageIQ::Providers::Amazon::CloudManager < ManageIQ::Providers::CloudMana
   require_nested :RefreshParser
   require_nested :RefreshWorker
   require_nested :Refresher
-  require_nested :SecurityGroup
   require_nested :Template
   require_nested :Vm
 
   OrchestrationTemplateCfn.register_eligible_manager(self)
   ExtManagementSystem.register_cloud_discovery_type('amazon' => 'ec2')
 
+  include ManageIQ::Providers::Amazon::ManagerMixin
+
+  has_one :network_manager,
+          :foreign_key => :parent_ems_id,
+          :class_name  => "ManageIQ::Providers::Amazon::NetworkManager",
+          :autosave    => true,
+          :dependent   => :destroy
+
+  delegate :floating_ips,
+           :security_groups,
+           :cloud_networks,
+           :cloud_subnets,
+           :network_ports,
+           :network_routers,
+           :public_networks,
+           :private_networks,
+           :all_cloud_networks,
+           :to        => :network_manager,
+           :allow_nil => true
+
+  before_validation :ensure_managers
+
+  def ensure_managers
+    build_network_manager unless network_manager
+    network_manager.name            = "#{name} Network Manager"
```

Comment:
shouldnt this be wrapped inside the `unless network_manager`? like
```ruby
def ensure_managers
    return if network_manager
    build_network_manager(
      name: "#{name} Network Manager"
    )
end
```

and tbh, I'd rather put this into a persistence callback e.g `before_create`